met-web/src/components/survey/create/CloneOptions.tsx

```diff
@@ -0,0 +1,232 @@
+
+const CloneOptions = () => {
+    const navigate = useNavigate();
+    const dispatch = useAppDispatch();
+    const [selectedSurvey, setSelectedSurvey] = useState<Survey | null>(null);
+    const [selectedEngagement, setSelectedEngagement] = useState<Engagement | null>(null);
+    const [loadingSurveys, setLoadingSurveys] = useState(true);
+    const [loadingEngagements, setLoadingEngagements] = useState(true);
+    const [isSaving, setIsSaving] = useState(false);
+
+    const {
+        surveyForm,
+        handleSurveyFormChange,
+        availableSurveys,
+        setAvailableSurveys,
+        availableEngagements,
+        setAvailableEngagements,
+    } = useContext(CreateSurveyContext);
+
+    const initialFormError = {
+        name: false,
+    };
+    const [formError, setFormError] = useState(initialFormError);
+
+    const handleChange = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+        handleSurveyFormChange({
+            ...surveyForm,
+            [e.target.name]: e.target.value,
+        });
+
+        if (hasKey(formError, e.target.name) && formError[e.target.name]) {
+            setFormError({
+                ...formError,
+                [e.target.name]: false,
+            });
+        }
+    };
+
+    const handleFetchSurveys = async (
+        page: number,
+        size: number,
+        sort_order: 'asc' | 'desc' | undefined,
+        search_text?: string,
+    ) => {
+        try {
+            const fetchedSurveys = await getSurveysPage({
+                page: page,
+                size: size,
+                sort_order: sort_order,
+                search_text: search_text,
+            });
+            setAvailableSurveys(fetchedSurveys.items);
+            setLoadingSurveys(false);
+        } catch (error) {
+            dispatch(openNotification({ severity: 'error', text: 'Error occurred while fetching available surveys' }));
+        }
+    };
+
+    const handleFetchEngagements = async (
+        page: number,
+        size: number,
+        sort_order: 'asc' | 'desc' | undefined,
+        search_text?: string,
+    ) => {
+        try {
+            const fetchedEngagements = await getEngagements({
+                page: page,
+                size: size,
+                sort_order: sort_order,
+                search_text: search_text,
+            });
+            console.log('FETCH!');
+            console.log(fetchedEngagements);
+            setAvailableEngagements(
+                fetchedEngagements.items.filter((engagement, index) => engagement.surveys.length !== 0),
+            );
+            setLoadingEngagements(false);
+        } catch (error) {
+            dispatch(
+                openNotification({ severity: 'error', text: 'Error occurred while fetching available engagements' }),
+            );
+        }
+    };
+
+    useEffect(() => {
+        if (!availableSurveys) {
+            handleFetchSurveys(1, 10, 'asc', '');
```

Comment:
we're always only calling for 10? what if there's more?

Reply:
Perhaps for now let it fetch all surveys or atleast specify a big number (2000) page size. Until we implement proper pagination and lazy loading for it. @saravanpa-aot thoughts?

Reply:
i think thats a better idea